### PR TITLE
Fix acos() behavior for zero/negative, and add tests

### LINKED ISSUE #74: acos() returns a value outside of [0,pi] for negative values and panics on zero
I passed in a slightly negative value, -0.09557510558, and got a very unexpected value near -1.5. The correct output should be near +1.6 (i.e. close to +90 degrees).

I had intended to pass in 0.0, with an expected output of +1.57 (+90 degrees), but when I tried that, I got a panic. (The approximation is dividing by zero.)

There are hints in the associated StackExchange article (referenced in acos.rs) that suggest a fix for this.

### PR DESCRIPTION
**Pull request by timboldt**
Note: I'm still relatively new at Rust, so feedback on style is welcome.

The problem with acos() was that it didn't handle the singularity at zero, and it didn't normalize negative numbers. I made the behavior match f32:acos().

~~The only remaining concern I have is that there might still be an underflow very, very close to zero.~~

Closes #74

### CODE REVIEW

File: src/float/acos.rs [hunk 1]
```diff
@@ -4,24 +4,65 @@
...
+use core::f32::consts::PI;
 
 impl F32 {
     /// Computes `acos(x)` approximation in radians in the range `[0, pi]`.
     pub(crate) fn acos(self) -> Self {
-        ((Self::ONE - self * self).sqrt() / self).atan()
+        if self > 0.0 {
+            ((Self::ONE - self * self).sqrt() / self).atan()
+        } else if self == 0.0 {
+            F32(PI / 2.0)
```
Comment (tarcieri) [inline src/float/acos.rs]: For consistency with the other ops I'd suggest:

```suggestion
            Self(PI / 2.0)
```

**Comment by tarcieri** [comment]
> The only remaining concern I have is that there might still be an underflow very, very close to zero.

Perhaps add test vectors for `f32::EPSILON` and `-f32::EPSILON`?

**Comment by timboldt** [comment]
> Perhaps add test vectors for f32::EPSILON and -f32::EPSILON?

Done.

**Comment by tarcieri** [comment]
Awesome, thank you!

**Pull request merged**